numba/core/utils.py

```diff
@@ -296,26 +296,66 @@ def __setitem__(self, key, value):
         super(UniqueDict, self).__setitem__(key, value)
 
 
-# Django's cached_property
-# see https://docs.djangoproject.com/en/dev/ref/utils/#django.utils.functional.cached_property    # noqa: E501
-
-class cached_property(object):
-    """
-    Decorator that converts a method with a single self argument into a
-    property cached on the instance.
-
-    Optional ``name`` argument allows you to make cached properties of other
-    methods. (e.g.  url = cached_property(get_absolute_url, name='url') )
-    """
-    def __init__(self, func, name=None):
-        self.func = func
-        self.name = name or func.__name__
-
-    def __get__(self, instance, type=None):
-        if instance is None:
-            return self
-        res = instance.__dict__[self.name] = self.func(instance)
-        return res
+if PYVERSION > (3, 7):
+    from functools import cached_property
+else:
+    from threading import RLock
+
+    # The following cached_property() implementation is adapted from CPython:
+    # https://github.com/python/cpython/blob/3.8/Lib/functools.py#L924-L976
```

Comment:
Explicit SHA?

Reply:
added in https://github.com/numba/numba/pull/7003/commits/15f9a1f28ce6f3d78ce9716166cae221c2c0dd38

Reply:
thanks